src/gt4py/next/otf/compilation/compiler.py

```diff
@@ -85,10 +85,17 @@ def __call__(
                 f"On-the-fly compilation unsuccessful for '{inp.program_source.entry_point.name}'."
             )
 
-        return getattr(
+        compiled_prog = getattr(
             importer.import_from_path(src_dir / new_data.module), new_data.entry_point_name
         )
 
+        @dataclasses.dataclass(frozen=True)
+        class Wrapper(stages.ExtendedCompiledProgram):
+            implicit_domain: bool = inp.program_source.implicit_domain
+            __call__: stages.CompiledProgram = compiled_prog
+
+        return Wrapper()
```

Comment:
This is to propagate the information whether the compiler needs to put size args generation into it's decorate step (if `implicit_domain == True`) or not (in which case explicit domain size args are part of the program's signature).